client/testing.go

```diff
@@ -91,34 +92,41 @@ func TestClientWithRPCs(t testing.T, cb func(c *config.Config), rpcs map[string]
 // with the server and then returns mock RPC responses for those interfaces
 // passed in the `rpcs` parameter. Useful for testing client RPCs from the
 // server. Returns the Client, a shutdown function, and any error.
-func TestRPCOnlyClient(t testing.T, srvAddr net.Addr, rpcs map[string]interface{}) (*Client, func() error, error) {
+func TestRPCOnlyClient(t testing.T, cb func(c *config.Config), srvAddr net.Addr, rpcs map[string]any) (*Client, func() error, error) {
```

Comment:
the cleanup cancelFunc being a `func() error` seems a little odd for a test helper, which could instead `t.Error()`. also it looks like all the callers in `client_csi_endpoint_test.go` at least are discarding the error anyway?  maybe it doesn't really matter, but that could possibly leak the cleanup goroutine without us knowing.

Reply:
Yeah good call -- we can make this return `(*Client, func())` and have it `must.NoError(t, err)` internally for its own errors as well.

Reply:
I tend to do `test.NoError(t, err)` instead of `must` in helper cleanup, because
 * other things after the cleanup may still be attempted (such as other cleanup)
 * [`t.FailNow`](https://pkg.go.dev/testing#T.FailNow) (via `must` -> `t.Fatalf`) can have odd behavior when run in goroutines.

Reply:
Normally I'd 100% agree with you, but this is setup code... if it doesn't run we shouldn't continue because we're likely to crash the test process.